Sources/Xcodeproj/Module+PBXProj.swift

```diff
@@ -298,3 +291,21 @@ extension Module  {
         return buildSettings
     }
 }
+
+extension Plist {
+    // FIXME: This is only internal for unit testing.
+    /// Quotes the strings in the Plist.
+    func quoteStrings() -> Plist {
+        switch self {
+        case .string(let str):
+            return .string("\"" + str + "\"")
```

Comment:
To clarify a bit:  there are a couple of different layers here.  At the bottom is the conversion of the plist node tree to a sequence of bytes.  There is quoting going on there to make sure that any type of string contents can be represented.  And where there is quoting there is also escaping:  by assigning special meaning to quotes, there becomes a need to escape quotes that shouldn't have special meaning.  And then also to escape the escape so that it doesn't have special meaning either.  :)

Then the next layer up is what's put into the plist nodes.  If what is being put into a plist node is a string build setting value, then no quoting is needed.  If it is an array build setting value, though, then it will need to be turned into a string by quoting each array element and joining the elements with whitespace.  Note that this decision (of which build settings get quoted) does not depend on what characters are in the value (e.g. whitespace), but rather on whether the build setting is an array or a string.